javascript/javascript1/week4/creditcardformatter.js

```diff
@@ -0,0 +1,18 @@
+function formatCreditCardNumber(number) {
+  if (typeof number == "string") {
+    console.log("Enter a valid card number");
+  } 
+  else {
+     number = number.toString();
+     const spaceBetweenDigits = 4;
+     let result = " ";
+     while (number.length > 0) {
```

Comment:
i think you can also do this simpler by using string.join().

Reply:
function formatCreditCardNumber(number) {
    const string = number.toString();
    const array = [];
    for (let i = 0; i < string.length; i += 4) {
        array.push(string.slice(i, i + 4));
    }
    return array.join(' ');
}

const formattedCreditCardObject = formatCreditCardNumber(123456789);
console.log(formattedCreditCardObject);